### PR TITLE
chore: angular upgrade from 11 to 12

### PR DESCRIPTION
**Pull request by itssharmasandeep**
## Description
Angular upgrade from 11 to 12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "test": "ng test",
     "lint": "ng lint",
     "commit": "git-cz",
-    "build:ci": "ng build --prod hyperdash-angular && ng build --prod hyperdash-angular-app",
+    "build:ci": "ng build --configuration production hyperdash-angular && ng build --configuration production hyperdash-angular-app",
```
Comment (anandtiwary) [inline package.json:11]: do we need to add a configuration flag here? I think Angular moving towards prod build by default. I would say let opt in for this feature. 

@aaron-steinfeld thoughts?

```
You can run the optional migration for enabling production builds by default ng update @angular/cli@12 --migrate-only production-by-default.

```

Comment (anandtiwary) [reply package.json:11]: We should also enable `NG_PERSISTENT_BUILD_CACHE` for faster build.

https://github.com/angular/angular-cli/pull/20756

```
  "start": "NG_PERSISTENT_BUILD_CACHE=1 ng serve",
  "build": "NG_PERSISTENT_BUILD_CACHE=1 ng build",
```

File: package.json [hunk 2]
```diff
@@ -17,14 +17,14 @@
...
-    "@angular/platform-browser-dynamic": "^11.1.2",
-    "@angular/router": "^11.1.2",
+    "@angular/animations": "^12.2.1",
+    "@angular/common": "^12.2.1",
+    "@angular/compiler": "^12.2.1",
+    "@angular/core": "^12.2.1",
+    "@angular/forms": "^12.2.1",
+    "@angular/platform-browser": "^12.2.1",
+    "@angular/platform-browser-dynamic": "^12.2.1",
+    "@angular/router": "^12.2.1",
```
Comment (anandtiwary) [inline package.json:27]: This should bump zone.js too right?

**Comment by anandtiwary** [comment]
Don't forget to update peer dependencies in hyperdash angular lib.

**Comment by anandtiwary** [comment]
@aaron-steinfeld Please take a look

File: package.json (hunk 1 above)
Comment (aaron-steinfeld) [reply package.json:11]: > I would say let's opt in for this feature.

So we can certainly compare build times, but does this mean our local builds would be minified, not source mapped etc? If so, I'd be against it. In other words, what is the advantage of doing a prod build locally other than a greater fidelity to the final state?

Comment (anandtiwary) [reply package.json:11]: They have better template checking. I would also not want to miss out on source map.

**Comment by itssharmasandeep** [comment]
Some issue

File: package.json (hunk 2 above)
Comment (itssharmasandeep) [reply package.json:27]: This is on latest version https://www.npmjs.com/package/zone.js?activeTab=versions

**Pull request closed without merging**